src/orders/components/OrderTransaction/components/CardTitle/Title.tsx

```diff
@@ -0,0 +1,88 @@
+import { ButtonLink } from "@dashboard/components/ButtonLink";
+import { TransactionItemFragment } from "@dashboard/graphql";
+import { capitalizeEachWord } from "@dashboard/misc";
+import { FakeTransaction } from "@dashboard/orders/types";
+import { ExternalLinkIcon, Text } from "@saleor/macaw-ui-next";
+import React from "react";
+import { IntlShape } from "react-intl";
+
+import { getInitiatorDisplayName } from "../../utils";
+import { PspReference } from "../TransactionEvents/components";
+
+const TitleLeftComponent = ({
+  transaction,
+  intl,
+  idx,
+}: {
+  transaction: TransactionItemFragment | FakeTransaction;
+  idx: number;
+  intl: IntlShape;
+}) => {
+  const isLeftSideLink = transaction.externalUrl && !transaction.pspReference;
+  const initiator = getInitiatorDisplayName(transaction);
+
+  let title = transaction.name;
+
+  if (initiator) {
+    const initiatorTitle = capitalizeEachWord(initiator).trim();
+    const transactionName = title ? `: ${title}` : "";
+    const colon = transaction.pspReference ? ": " : "";
+
+    title = `${initiatorTitle}${transactionName}${colon}`;
+  } else {
+    title =
+      title ||
+      intl.formatMessage(
+        {
+          defaultMessage: "Transaction No. {idx}",
+          id: "s4Y+0N",
+          description: "transaction title fallback",
+        },
+        {
+          idx,
+        },
+      );
```

Comment:
i'd move this to separate messages file
```suggestion
      intl.formatMessage(
        {
          defaultMessage: "Transaction No. {idx}",
          id: "s4Y+0N",
          description: "transaction title fallback",
        },
        {
          idx,
        },
      );
```